PLUGINS.md

```diff
@@ -1,54 +1,153 @@
+
+* A `package.json` file which contains the following metadata: name, version, supported runtime versions, dependencies and others. For more information, see the [Package.json Specification](#package-json-specification) section.
+* One or more CommonJS modules that expose a native API via a single JavaScript API. For more information about Common JS modules, see the [CommonJS Wiki](http://wiki.commonjs.org/wiki/CommonJS).
+* `AndroidManifest.xml` and `Info.plist` which describe the permissions and features required or used by your app for Android and iOS, respectively.
+
+The plugin must have the directory structure, described in the [Directory Structure](#directory-structure) section.
+
+## Create a Plugin
+
+If the NativeScript framework does not expose a native API that you need, you can develop a custom plugin which exposes the required functionality. When you develop a custom plugin, keep in mind the following requirements.
+
+* The plugin must be a valid npm package.
+* The plugin must expose a built-in native API. Currently, you cannot expose native APIs available via custom native libraries.
+* The plugin must be written in JavaScript and must comply with the CommonJS specification.
+* The plugin directory structure must comply with the specification described below.
+* The plugin must contain a valid `package.json` which complies with the specification described below.
+* The plugin must contain `AndroidManifest.xml` and `Info.plist` file which describe any platform-specific settings such as required permissions.
+
+### Directory Structure
+
+NativeScript plugins which consist of one CommonJS module must have the following directory structure.
 
-A package json file has the following structure:
 ```
-{
-	"name": "myplugin",
-	"version": "0.0.1",
-	"nativescript": {
-		"platforms": {
-			"ios": "1.0.0",
-			"android": "1.1.0"
-		}
-	}
-}
+MyPlugin/
+├── index.js
+├── package.json
+└── platforms/
+    ├── android/
+    │   └── AndroidManifest.xml
+    └── ios
+        └── Info.plist
 ```
-`nativescript` key is a mandatory section for a NativeScript plugin. `platforms` section describes the supported runtime versions for each platform.
 
-A plugin directory has the following structure:
+NativeScript plugins which consist of multiple CommonJS modules must have the following directory structure.
+
 ```
-myPlugin/
--- index.js
--- package.json
--- platforms
-  -- Android
-	   -- myjar.jar
-	   -- Android.manifest
-  -- iOS
-	   -- myFramework.framework
-	   -- Info.plist
+MyPlugin/
+├── MyModule1/
+│   ├── index1.js
+│   ├── package.json
+│   └── platforms/
+│       ├── android/
+│       │   └── AndroidManifest.xml
+│       └── ios
+│           └── Info.plist
+└── MyModule2/
+    ├── index2.js
+    ├── package.json
+    └── platforms/
+        ├── android/
+        │   └── AndroidManifest.xml
+        └── ios
+            └── Info.plist
 ```
```

Comment:
It is not possible to have 2 `platforms`  folders into one plugin. Also `platforms` folder should be at root level.